src/mail/mail.service.ts

```diff
@@ -27,14 +37,38 @@ export class MailService {
         this.hostname = `${this.configService.getOrThrow('APP_HOST')}:${this.configService.getOrThrow('APP_PORT')}`;
     }
 
-    async sendUserConfirmation(user: User, token: string) {
+    async sendUserConfirmation(email: string, token: string) {
         const url = `${this.hostname}/auth/confirm?token=${token}`;
         const template = compile(EMAIL_CONFIRMATION_TEMPLATE);
-    
-        await this.mailerService.sendMail({
-            to: user.email,
+
+        return await this.mailerService.sendMail({
```

Comment:
Normally, you shouldn't do both `await` and `then/catch` right? You use Promises with `await` to avoid using `then`. 
Maybe just do `this.mailerService.sendMail(... then(... return..., catch(... return...`, without `return await`?

Reply:
Good point, I'll change that !